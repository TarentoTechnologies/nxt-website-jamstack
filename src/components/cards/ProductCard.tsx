import { Link } from "gatsby";

import { bgUraniaBlue } from "../../styles/style-guide/ColorGuide.module.css";
import { cardHeader } from "../../styles/style-guide/Typography.module.css";
import {
  customBgStyle,
  logoStyles,
  proCard,
  proCardSecondary,
  secondaryFontsize,
  titleStyles,
} from "./ProductCard.module.css";

interface ProductCardProps {
  title: string;
  bgImg?: any;
  logo?: any;
  isSecondary?: boolean;
  secLogo?: any;
  link?: string;
  isLink?: boolean;
  customBgPosition?: boolean;
}

export const ProductCard = ({
  title,
  bgImg,
  logo,
  secLogo,
  isSecondary,
  link,
  isLink,
  customBgPosition,
}: ProductCardProps) => {

  if (isLink) {
    return (
      <Link to={link != null ? link : ""}>
        <div
          className={`container-fluid ${
            isSecondary ? proCardSecondary : proCard
          } ${customBgPosition ? `${bgUraniaBlue} ${customBgStyle}` : ""}`}
          style={{ backgroundImage: `url(${bgImg})` }}
        >
          <div
            className={`${titleStyles} ${cardHeader} ${
              isSecondary ? secondaryFontsize : ""
            }`}
          >
            <span data-aos="fade-up">{title}</span>
          </div>
          {(secLogo || logo) && (
            <div className={`${logoStyles}`}>
              <img
                src={isSecondary ? secLogo : logo}
                alt={title}
                title={title}
                width={120}
                data-aos="fade-up"
              />
            </div>
          )}
        </div>
      </Link>
    );
  } else {
    return (
      <div
        className={`container-fluid ${
          isSecondary ? proCardSecondary : proCard
        } ${customBgPosition ? `${bgUraniaBlue} ${customBgStyle}` : ""}`}
        style={{ backgroundImage: `url(${bgImg})` }}
      >
        <div
          className={`${titleStyles} ${cardHeader} ${
            isSecondary ? secondaryFontsize : ""
          }`}
        >
          <span data-aos="fade-up">{title}</span>
        </div>
        {(secLogo || logo) && (
          <div className={`${logoStyles}`}>
            <img
              src={isSecondary ? secLogo : logo}
              alt={title}
              title={title}
              width={80}
              data-aos="fade-up"
            />
          </div>
        )}
      </div>
    );
  }
};
