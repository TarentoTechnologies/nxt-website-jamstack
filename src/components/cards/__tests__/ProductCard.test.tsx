import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { ProductCard } from "../ProductCard";

describe("Test if the ProductCard component is working as expected", () => {
  it("<ProductCard /> matches snapshot", () => {
    const component = render(
      <ProductCard
        title={"Bolt - Data platform"}
        bgImg={
          "https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_2f3dc77aa7.png"
        }
        logo={
          "https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_dark_94a1e2c631.svg"
        }
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<ProductCard /> renders without exploding", () => {
    const component = render(
      <ProductCard
        title={"Bolt - Data platform"}
        bgImg={
          "https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_2f3dc77aa7.png"
        }
        logo={
          "https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_dark_94a1e2c631.svg"
        }
      />
    );
    expect(component.getByTestId("product-card")).toBeInTheDocument();
  });
});
