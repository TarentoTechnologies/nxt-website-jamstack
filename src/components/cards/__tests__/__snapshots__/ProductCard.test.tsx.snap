// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the ProductCard component is working as expected <ProductCard /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid proCard "
    data-testid="product-card"
    style="background-image: url(https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_2f3dc77aa7.png);"
  >
    <div
      class="titleStyles cardHeader "
    >
      <span
        data-aos="fade-up"
      >
        Bolt - Data platform
      </span>
    </div>
    <div
      class="logoStyles"
    >
      <img
        alt="Bolt - Data platform"
        data-aos="fade-up"
        src="https://strapi-stage-nxt.tarento.com/uploads/tech_bolt_dark_94a1e2c631.svg"
        title="Bolt - Data platform"
        width="80"
      />
    </div>
  </div>
</div>
`;
